base/loading.jl

```diff
@@ -772,6 +780,17 @@ function stale_cachefile(modpath::String, cachefile::String)
                 return true
             end
         end
+
+        # finally, verify that the cache file has a valid checksum
+        data = Mmap.mmap(io, Vector{UInt8}, filesize(io), 0)
+        checksum = ntoh(reinterpret(UInt32, data[end-3:end])[1])
```

Comment:
Why is it definitely not aligned?  `data[end-3:end]` allocates a new array.

Anyway, I can fix it to not use `reinterpret`.